sdk/core/Azure.Core.TestFramework/src/RecordSession.cs

```diff
@@ -89,7 +92,14 @@ public RecordEntry Lookup(RecordEntry requestEntry, RecordMatcher matcher, Recor
                 Entries.Remove(entry);
                 return entry;
             }
+        }
 
+        private void NormalizeLineEndings(RecordEntryMessage entryMessage)
+        {
+            if (entryMessage.TryGetBodyAsText(out string text))
+            {
+                entryMessage.Body = Encoding.UTF8.GetBytes(text.Replace("\n", "\r\n"));
+            }
```

Comment:
Need to either check that environment.newline is \n or use regex to ensure \n is not preceded by \r